packages/yarnpkg-shell/sources/errors.ts

```diff
@@ -0,0 +1,14 @@
+export type ShellErrorOptions = {
+  recoverable: boolean;
+};
+
+export class ShellError extends Error {
+  public recoverable: boolean;
```

Comment:
Why not just keep using the regular `Error` class for unrecoverable errors, instead of keeping a `recoverable` flag?

Reply:
Because I wanted shell errors to have a `ShellError` name instead of just `Error`. Do you think it makes sense or should I remove it?

Reply:
Hm .. I'd say that imo the extra complexity isn't worth it 🤔

Reply:
Makes sense! I'll change it 👍